### ISSUE TITLE
Turn options from from strings into objects?

### ISSUE THREAD

**Issue by janosh** [janosh/svelte-multiselect#13]
Considering a move from a simple string array as options to objects with keys like

```ts
type Option = {
  label: string
  value: string | number
  title: string // tooltip to display on hovering this option
  disabled: boolean
  disabledTitle: string // overrides the fallback disabledTitle = 'This option is disabled'
  selectedTitle: string // tooltip to display when this option is selected and hovered
  // maybe more?
}
```

Posting this here to gather thoughts and solicit feedback.

**Comment by MelpomeneMU** [comment]
Yes, please! This would make it truly ready for primetime.

I've got multiple use cases dealing with DB data where the option text should be different from the value passed to an API. An example: six different locations with identical names, all in different places, each with a different ID... you'd better believe we go by the ID!

You might also consider a template of some sort for options. In my example case, options would be formatted something like this:

<Option>
[Location name] ([customer friendly ID])
[address line 1]
[city state zip]
<Button text="Remove">
</Option>

Something that customizable  would come in very handy.

**Comment by janosh** [comment]
@MelpomeneMU Thanks for the input! Considering this seems to be quite a popular feature, I'll try to start working on it this weekend.

Just to clarify your example, if there are six identically named places in different locations, you must already include some additional information (like country) for your users, right? Else how would they know which of the 6 to pick?

In any case, I do see the value of sending a different value than what is displayed to the user to an API.

Regarding the template string, I believe it would be easiest for devs to do that on their own. Most people just want to render simple strings, so seems overkill to ship the logic required to hydrate a template to everyone regardless of whether they need it or not. It should as simple as running a `map()` over your array of options with some string interpolation to build the template you need yourself.

On a related note, we might consider parsing labels as HTML though with Svelte's `{@html ...}` directive in case people want to have tags with inline styles or some such.

**Comment by janosh** [comment]
Ok, I think I have something I would tentatively call a beta version of options as objects. Still a few TS issues to sort out. Will try to do some testing tomorrow.

**Comment by janosh** [comment]
Turned out to be a bigger refactor than expected. I pushed everything to the [`object-options`](janosh/svelte-multiselect/tree/object-options) branch. No known issues so would be great if everyone interested in this feature could give it a try and report bugs/suggestions before I release this as `v2.0.0`.

```sh
yarn add yarn add https://github.com/janosh/svelte-multiselect#object-options
```

**Comment by janosh** [comment]
This is released in [v2.0.0](https://www.npmjs.com/package/svelte-multiselect/v/2.0.0). 🎉 

Note the breaking changes mentioned in #16, mainly that `bind:selected` now returns objects instead of strings. To get the strings as before, use `bind:selectedLabels`.
